CDCgov/prime-reportstream
# 10774 Refactor and increase coverage for BlobAccess

This PR increases code coverage for BlobAccess and implements reuse of Azure blob container clients.

Test Steps:
1. Confirm [code coverage report](https://sonarcloud.io/code?id=CDCgov_prime-data-hub&pullRequest=10732&selected=CDCgov_prime-data-hub%3Aprime-router%2Fsrc%2Fmain%2Fkotlin%2Fazure%2FBlobAccess.kt) is over 90%.
2. Ensure all tests run successfully.

## Changes
- Store Azure container client after creation so that subsequent calls can reuse it.
- Isolated function to return Blob connection string (needed for mocking)
- Added unit tests.

## Checklist

### Testing

## Linked Issues
- #10774
